### PR TITLE
[SW-2277] Update GridSearch Documentation

### PR DESCRIPTION
**Pull request by mn-mikke**

### CODE REVIEW

File: ml/src/test/scala/ai/h2o/sparkling/ml/algos/H2OGridSearchTestSuite.scala [hunk 1]
```diff
@@ -155,6 +154,7 @@ class H2OGridSearchTestSuite extends FunSuite with Matchers with SharedH2OTestCo
       .setLabelCol("AGE")
       .setHyperParameters(hyperParams)
       .setAlgo(algo)
+      .setStrategy()
```
Comment (jakubhava) [inline ml/src/test/scala/ai/h2o/sparkling/ml/algos/H2OGridSearchTestSuite.scala:157]: Argument missing to setStrategy

File: doc/src/site/sphinx/tutorials/sw_grid.rst [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+            import scala.collection.mutable.HashMap
+            val hyperSpace: HashMap[String, Array[AnyRef]] = HashMap()
+            hyperSpace += "ntrees" -> Array(1, 10, 30).map(_.asInstanceOf[AnyRef])
+            hyperSpace += "mtries" -> Array(-1, 5, 10).map(_.asInstanceOf[AnyRef])
+
+        Pass the algorithm and hyper space to the grid search and set properties defining tha way how the hyper space will be traversed.
+
+        Sparkling Water supports two strategies for traversing hyperspace:
+
+        - **Cartesian** - (Default) This strategies tries out every possible combination of hyper parameter values and
```
Comment (jakubhava) [inline doc/src/site/sphinx/tutorials/sw_grid.rst:61]: typo: strategies -> strategy

File: doc/src/site/sphinx/tutorials/sw_grid.rst [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+          finishes after the whole space is traversed.
+        - **RandomDiscrete** - In each iteration, the strategy randomly selects the combination of values from the hyper space and
+          can be terminated before the whole space is traversed. The termination could be dependent on various criteria
```
Comment (jakubhava) [inline doc/src/site/sphinx/tutorials/sw_grid.rst:64]: `could be dependent` -> `depends`

File: doc/src/site/sphinx/tutorials/sw_grid.rst [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+        .. code:: python
+
+            hyperSpace = {"ntrees": [1, 10, 30], "mtries": [-1, 5, 10]}
+
+        Pass the algorithm and hyper space to the grid search and set properties defining tha way how the hyper space will be traversed.
+
+        Sparkling Water supports two strategies for traversing hyperspace:
+
+        - **Cartesian** - (Default) This strategies tries out every possible combination of hyper parameter values and
```
Comment (jakubhava) [inline doc/src/site/sphinx/tutorials/sw_grid.rst:138]: DTTO

File: doc/src/site/sphinx/tutorials/sw_grid.rst [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+          finishes after the whole space is traversed.
+        - **RandomDiscrete** - In each iteration, the strategy randomly selects the combination of values from the hyper space and
+          can be terminated before the whole space is traversed. The termination could be dependent on various criteria
```
Comment (jakubhava) [inline doc/src/site/sphinx/tutorials/sw_grid.rst:141]: DTTO

File: ml/src/test/scala/ai/h2o/sparkling/ml/algos/H2OGridSearchTestSuite.scala (hunk 1 above)
Comment (mn-mikke) [reply ml/src/test/scala/ai/h2o/sparkling/ml/algos/H2OGridSearchTestSuite.scala:157]: oh, good catch, this file shouldn't have been committed

**Pull request merged**